src/transformers/models/llava/processing_llava.py

```diff
@@ -45,6 +48,10 @@ class LlavaProcessor(ProcessorMixin):
 
     def __init__(self, image_processor=None, tokenizer=None):
         super().__init__(image_processor, tokenizer)
+        self.image_size = self.image_processor.size["shortest_edge"]
```

Comment:
As llava can take any image processor, we can't assume `size` will have `shortest_edge`, it should also be a dict of `{"height": h, "width": w}`

Reply:
since we can't be sure which one is used by image processor and what's the output size, I decided to check `pixel_values` size and use that as `self.image_size`

Also, now we don't assume it's always a square size output and calculate patches for height and width separately, similar to what ViT does in its modeliing file.